pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -99,6 +99,16 @@ func NewInstanceSettings(cfg *setting.Cfg, clientProvider *httpclient.Provider,
 			return nil, fmt.Errorf("error getting credentials: %w", err)
 		}
 
+		customizedCloudSettings, err := getCustomizedCloudSettings(cloud, settings.JSONData)
+		if err != nil {
+			return nil, fmt.Errorf("error getting credentials: %w", err)
+		}
```

Comment:
nit: you can move this to getRoutes, there is no need to do this parsing if the cloud is not the customized one.

Reply:
Agree, in `getCustomizedCloudSettings()` no deserialization will happen if `cloud` is not `AzureCustomizedCloud`. Another consideration here is other fields might be added into the JSON data, so it should be better to keep this function then move all logic into `getAzureRoutes()`.

Reply:
I still think it's better to move this call to `getAzureRoutes`:

 - You don't need to do `if azureCustomizedCloud` twice, you can assume that when calling `getCustomizedCloudSettings`, we are in the "customized" case.
 - Other clouds are not affected by this change. If more fields are added to the JSONData, we can assume it will be for the customized case, we don't foresee changes in the existing clouds.

Reply:
Refactored.